Pcap++/header/DeviceUtils.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+/// @file
+
+#include <memory>
+#include "IpAddress.h"
+#include "MemoryUtils.h"
+
+// Forward declaration
+struct pcap_rmtauth;
+
+namespace pcpp
+{
+	namespace internal
+	{
+		/**
+		 * Fetches a list of all network devices on the local machine that LibPcap/WinPcap/NPcap can find.
+		 * @return A smart pointer to an interface list structure.
+		 * @throws std::runtime_error The system encountered an error fetching the devices.
+		 */
+		std::unique_ptr<pcap_if_t, PcapFreeAllDevsDeleter> getAllLocalPcapDevices();
+#ifdef _WIN32
+		/**
+		 * Fetches a list of all network devices on a remote machine that WinPcap/NPcap can find.
+		 * @param[in] ipAddress IP address of the remote machine.
+		 * @param[in] port Port to use when connecting to the remote machine.
+		 * @param[in] pRmAuth Pointer to an authentication structure to use when connecting to the remote machine. Nullptr if no authentication is required.
+		 * @return A smart pointer to an interface list structure.
+		 * @throws std::runtime_error The system encountered an error fetching the devices.
+		 */
+		std::unique_ptr<pcap_if_t, PcapFreeAllDevsDeleter> getAllRemotePcapDevices(const IPAddress& ipAddress, uint16_t port, pcap_rmtauth* pRmAuth = nullptr);
+#endif // _WIN32
```

Comment:
This is very specific to remote devices, why do we need to extract it to a different header file? 🤔

Reply:
Mostly so all the pcap fetch devices calls are in the same place.

Reply:
to be honest, remote pcap devices are quite a small niche and I don't think it's used a lot. I'd recommend keeping the code inside `PcapRemoteDevice`, and keep these util files mostly focused on common code that is used in several places

Reply:
IDK, I find it better organized to have all the code that "gets pcap to fetch a list of X devices" in one place, even if this specific function that is not used in a lot of places.

@tigercosmos Any thoughts on the matter?

Reply:
I agree with @seladb , unless this function is used somewhere else except for `PcapRemoteDevice` in our codebase.

Reply:
Ok, moved the function to an anonymous namespace in `PcapRemoteDeviceList` as it is only used there.

Although I think I spotted an issue with `PcapRemoteDevice`'s `clone` method. It reuses the the `PcapLiveDevice`'s clone which fetches `getAllLocalDevices`. Btw, why is clone reusing re-fetching the devices instead of just copying the device data?